benchmark/ext_helpers/kmeans_dpcpp.py

```diff
@@ -0,0 +1,48 @@
+import dpctl.tensor as dpt
+
+from sklearn.exceptions import NotSupportedByEngineError
+
+from sklearn_numba_dpex.kmeans.engine import KMeansEngine
+
+import kmeans_dpcpp as kdp
+
+
+class KMeansDPCPPEngine(KMeansEngine):
+    def init_centroids(self, X):
+        init = self.init
+        if isinstance(init, str) and init == "k-means++":
+            raise NotSupportedByEngineError(
+                "kmeans_dpcpp does not support k-means++ init."
+            )
+        return super().init_centroids(X)
+
+    def kmeans_single(self, X, sample_weight, centers_init_t):
+        n_samples, n_features = X.shape
+        device = X.device.sycl_device
+
+        assignments_idx = dpt.empty(n_samples, dtype=dpt.int32, device=device)
+        res_centroids_t = dpt.empty_like(centers_init_t)
+
+        # TODO: make work_group_size and centroid window dimension consistent in
+        # kmeans_dpcpp with up to date configuration in sklearn_numba_dpex
+        centroids_window_height = 8
+        work_group_size = 128
+
+        centroids_private_copies_max_cache_occupancy = 0.7
```

Comment:
How has this value for `centroids_private_copies_max_cache_occupancy` been set?

Reply:
Arbitrarily. The idea is "have the max possible number of copies in memory of the centroids array, but do not saturate more than x% of the cache, in case something else needs it". I think I played with this value and didn't notice telling differences on performance.

Reply:
Thank for following-up on this question. I think it would be nice to document the motivation for introducing `centroids_private_copies_max_cache_occupancy` in the implementations. I think this better be done in another PR.

What do you think?

Reply:
ok, opening a todo for my future self then.